src/main/java/net/snowflake/client/core/IncidentUtil.java

```diff
@@ -239,10 +239,10 @@ private static void writeVmMetrics(JsonGenerator json, VirtualMachineMetrics vm)
    * @return the given Throwable object
    */
   public static Throwable generateIncidentV2WithException(
-      SFSession session, Throwable exc, String jobId, String requestId) {
+      SFSessionInterface session, Throwable exc, String jobId, String requestId) {
     // Generate an incident only if the session exists.
     if (session != null) {
-      new Incident(session, exc, jobId, requestId).trigger();
+      session.raiseError(exc, jobId, requestId);
```

Comment:
Why is this necessary ?

Reply:
@sfc-gh-sshankar `Incident` will try to log an incident with Snowflake and GS. We wish to disable this code path and have the session be able to implement its own error handler.

We could of course, make `Incident` more generic so as to disable the network egress, just as an alternative approach.